src/w_wad.c

```diff
@@ -91,90 +92,136 @@ static int *handles = NULL;
 
 static void W_AddFile(const char *name) // killough 1/31/98: static, const
 {
-  wadinfo_t   header;
-  lumpinfo_t* lump_p;
-  unsigned    i;
-  int         handle;
-  int         length;
-  int         startlump;
-  filelump_t  *fileinfo, *fileinfo2free=NULL; //killough
-  filelump_t  singleinfo;
-  boolean     is_single = false;
-  char        *filename = strcpy(malloc(strlen(name)+5), name);
-
-  NormalizeSlashes(AddDefaultExtension(filename, ".wad"));  // killough 11/98
-
-  // open the file and add to directory
-
-  if ((handle = M_open(filename,O_RDONLY | O_BINARY)) == -1)
+    wadinfo_t header;
+    filelump_t *fileinfo;
+    int handle;
+    int length;
+    int startlump;
+    char *filename;
+
+    if (M_StringCaseEndsWith(name, ".wad")
+        || M_StringCaseEndsWith(name, ".lmp"))
     {
-      if (strlen(name) > 4 && !strcasecmp(name+strlen(name)-4 , ".lmp" ))
-	{
-	  free(filename);
-	  return;
-	}
-      // killough 11/98: allow .lmp extension if none existed before
-      NormalizeSlashes(AddDefaultExtension(strcpy(filename, name), ".lmp"));
-      if ((handle = M_open(filename,O_RDONLY | O_BINARY)) == -1)
-	I_Error("Error: couldn't open %s\n",name);  // killough
+        filename = M_StringDuplicate(name);
     }
+    else
+    {
+        filename = M_StringJoin(name, ".wad", NULL);
```

Comment:
Do we need this? We've already looked for the path in `D_AddFile` using `D_TryFindWADByName`, so this works inconsistently. For example, on Windows, `-file av.wad` will work if `av.wad` is in DOOMWADDIR, but `-file av` will only work if `av.wad` is in the port directory.

Reply:
I'd like the `-file av` variant to work everywhere.

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/1578/commits/94b60c23a77f9e9df99eebe1140817257ae063bc